modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -1608,6 +1608,9 @@ public void commitTx(IgniteInternalTx tx) throws IgniteCheckedException {
                 ", tx=" + tx.getClass().getSimpleName() + ']');
         }
 
+        if (!cctx.kernalContext().clientNode())
```

Comment:
Can we have something like following instead of invoke `onCommit` for every commit. 

```
    private volatile Consumer<IgniteInternalTx> commitCb;
...
    public void commitTx(IgniteInternalTx tx) throws IgniteCheckedException {
        ....
        Consumer<IgniteInternalTx> commitCb0 = commitCb;
        if (commitCb0 != null)
            commitCb0.accept(tx);
    }
```

So `SnapshotManager` can set and clear callback when required.
And all other time we enlarge tx commit to one volatile null check.